src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/UserEntity.kt

```diff
@@ -316,8 +316,12 @@ enum class UserRole(val service: ServiceName, val cas1ApiValue: ApprovedPremises
   CAS1_EXCLUDED_FROM_ASSESS_ALLOCATION(ServiceName.approvedPremises, ApprovedPremisesUserRole.excludedFromAssessAllocation),
   CAS1_EXCLUDED_FROM_MATCH_ALLOCATION(ServiceName.approvedPremises, ApprovedPremisesUserRole.excludedFromMatchAllocation),
   CAS1_EXCLUDED_FROM_PLACEMENT_APPLICATION_ALLOCATION(ServiceName.approvedPremises, ApprovedPremisesUserRole.excludedFromPlacementApplicationAllocation),
-  CAS1_APPEALS_MANAGER(ServiceName.approvedPremises, ApprovedPremisesUserRole.appealsManager),
-  CAS1_JANITOR(ServiceName.approvedPremises, ApprovedPremisesUserRole.janitor),
+  CAS1_APPEALS_MANAGER(
+    ServiceName.approvedPremises,
+    ApprovedPremisesUserRole.appealsManager,
+    listOf(UserPermission.CAS1_PROCESS_AN_APPEAL, UserPermission.CAS1_VIEW_ASSESSMENTS, UserPermission.CAS1_ASSESS_APPEALED_APPLICATION),
+  ),
```

Comment:
I _think_ we may have a UI/API deployment race condition issue

If this API was deployed before UI changes we would allow appealed assessments to be allocated to to users with the role 'CAS1_APPEALS_MANAGER', but the UI wouldn't let them view assessments assigned to them

If the UI was deployed before the API there would be lots of errors because the API wouldn't be returning permissions (and users with existing roles won't be able to view the assessments anymore because they'll have no permissions!)

I think the answer is to use feature flags to determine whether the CAS1_APPEALS_MANAGER can be assigned appealed applications (i.e. they have the CAS1_ASSESS_APPEALED_APPLICATION permission). This will be disabled in production but enabled in all other environments (allowing us to do UAT).

Once we've starting using permissions across the UI this will no longer be an issue, but it'll just be some pain we'll need to deal with whilst migrating (And to be fair, this is the sort of issue we're introducing permissions to avoid!)

Let's have a chat about this when you're ready because it's awkward to inject a feature flag into enum construction so we'll need to add some indirection to do this. Also i have a PR to 'fix' feature flags so we'll need that merging first - https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/2094

Reply:
Resolved with introduction of `cas1-appeal-manager-can-assess-applications` feature flag, in UserService.